core/unit_test/tools/TestProfilingMacros.cpp

```diff
@@ -0,0 +1,12 @@
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Macros.hpp>
+#include <iostream>
+int main() {
+#ifdef KOKKOS_ENABLE_PROFILING_SOURCE_INFO
+  std::string expected = "TestProfilingMacros.cppKKSPLIT10KKSPLITdogs";
+#else
+  std::string expected = "dogs";
+#endif
+  std::string name = KOKKOS_FILELINE_NAME(dogs);
+  assert(name == expected);
```

Comment:
```suggestion
  struct Up {};
  if(name != expected) { throw Up(); }
```
PS you need to apply clang-format